neural_speed/core/layers/mha_dense.cpp

```diff
@@ -972,24 +1094,74 @@ class scale_track_max_t<ISA_T, float, float> {
       // if (j < utils::padto(N, 64))
       //   memset(dst + i * p.ld_dst + j, 0, sizeof(*dst) * (utils::padto(N, 64) - j));
     }
-#else
+    return BTLA_CODE::Success;
+  }
+#endif
+  template <bool HAS_ALIBI>
+  BTLA_CODE forward_(const SType* src, const int src_step, const int M_offset, const int N_offset, const int M,
+                     const int N, const Param& p) const {
+    const auto dst = p.dst + M_offset * p.ld_dst + N_offset;
+    const auto dst_max = p.dst_max + M_offset;
+#if MHA_2ND_EXP
+#if CompileAVX512F()
+    if constexpr (ISA_T >= BTLA_ISA::AVX512F) {
+      return forward_512<HAS_ALIBI>(src, src_step, M_offset, N_offset, M, N, p);
+    }
+#endif
+#if CompileAVX2()
+    if constexpr (ISA_T >= BTLA_ISA::AVX2) {
```

Comment:
why not ```forward_2<HAS_ALIBI>(src, src_step, M_offset, N_offset, M, N, p)``` too?

Reply:
Agree. (There was no `forward_512` before until I realized that there must be one to target different architecture).

Reply:
Refactored in 480d2871fd93ef5de42fa13202474c30c456cda4